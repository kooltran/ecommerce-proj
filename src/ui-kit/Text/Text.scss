@import '../styles/variables';

.text {
    margin-right: 5px;
    display: inline-block;
  
    &__text {
      color: $color-black;
      line-height: 19px;
    }
  
    &--bold {
      font-weight: 600;
    }
    &--italic {
      font-style: italic;
    }
  
    &--block {
      display: block;
    }
  
    &--white {
      color: $color-white;
    }

    &--link {
        cursor: pointer;
    }
}
  