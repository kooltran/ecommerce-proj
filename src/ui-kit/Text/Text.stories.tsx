import React from 'react'

import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import Text from './'

storiesOf('Conversation/UI-Kit/Text', module)
  .add(
    'Default',
    withInfo({ header: true })(() => (
      <Text>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry.{' '}
      </Text>
    ))
  )
  .add(
    'Bold',
    withInfo({ header: true })(() => (
      <Text isBold={true}>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry.
      </Text>
    ))
  )
  .add(
    'Italic',
    withInfo({ header: true })(() => (
      <Text isItalic={true}>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry.
      </Text>
    ))
  )
  .add(
    'with Heading',
    withInfo({ header: true })(() => (
      <>
        <Text isBold={true} isBlock={true}>
          Ciao
        </Text>

        <Text>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry.{' '}
        </Text>
      </>
    ))
  )
