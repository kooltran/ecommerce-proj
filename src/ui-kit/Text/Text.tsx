import * as React from 'react'
import classnames from 'classnames'
import './Text.scss'

export interface ITextProps {
    children: any
    className?: string
    isBlock?: boolean
    isBold?: boolean
    isLink?: boolean
    isItalic?: boolean
    onClick?: (e: any) => void
}

const Text: React.SFC<ITextProps> = ({
    isItalic,
    isBold,
    isLink,
    children,
    isBlock,
    onClick,
    className,
}) => {
    const styleText = classnames(className, {
        text__text: true,
        'text--bold': isBold,
        'text--italic': isItalic,
        'text--block': isBlock,
        'text--link': isLink,
    })

    return (
        <span
            className={`text ${styleText}`}
            onClick={onClick ? onClick : () => null}
        >
            {children}
        </span>
    )
}

export default Text

Text.displayName = 'Text'