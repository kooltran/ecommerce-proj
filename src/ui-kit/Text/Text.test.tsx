import React from 'react'
import renderer from 'react-test-renderer'
import Text from './Text'

it('should render', () => {
  const tree = renderer.create(<Text>Lorem Ipsum is simply dummy text of the printing and typesetting
    industry.</Text>).toJSON()
  expect(tree).toMatchSnapshot()
})
