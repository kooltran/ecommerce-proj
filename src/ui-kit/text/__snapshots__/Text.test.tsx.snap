// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<span
  className="text text__text"
  onClick={[Function]}
>
  Test
</span>
`;
